common/src/main/scala/uk/ac/wellcome/sqs/SQSReader.scala

```diff
@@ -39,4 +43,31 @@ class SQSReader @Inject()(sqsClient: AmazonSQS, sqsConfig: SQSConfig)
       .getMessages
       .toList
   }
+
+  private def processAndDeleteMessages(
+    messages: List[Message],
+    process: Message => Unit) = {
+      Future.sequence(messages.map { message =>
+        Future.fromTry(Try(process(message))
+          .recover {
+            case e: Throwable =>
+              error(s"Error processing message", e)
+              throw e
+        }).flatMap(_ => deleteMessage(message))
+      })
+  }
+
+  private def deleteMessage(message: Message) =
```

Comment:
Right, I think I added it in SQSLocal, but maybe that's not the right place, it's too hidden